File: opendental5.7/OpenDentBusiness/TableTypes/MountItem.cs
using System;
using System.Collections.Generic;
using System.Text;
using OpenDental.DataAccess;

namespace OpenDentBusiness {
	/// <summary>These are always attached to a mount and are constant. Should not be deleted, but rather updated if geometry changes.</summary>
	[DataObject("mountitem")]
	public class MountItem : DataObjectBase {
		///<summary>Primary key.</summary>
		[DataField("MountItemNum", PrimaryKey=true, AutoNumber=true)]
		private int mountItemNum;

		/// <summary>Primary key.</summary>
		public int MountItemNum {
			get { return mountItemNum; }
			set { mountItemNum = value; MarkDirty(); mountItemNumChanged = true; }
		}

		bool mountItemNumChanged;

		public bool MountItemNumChanged {
			get { return mountItemNumChanged; }
			
		}
		///<summary>FK to mount.MountNum.</summary>
		[DataField("MountNum")]
		private int mountNum;

		/// <summary>FK to mount.MountNum.</summary>
		public int MountNum {
			get { return mountNum; }
			set { mountNum = value; MarkDirty(); mountNumChanged = true; }
		}

		bool mountNumChanged;

		public bool MountNumChanged {
			get { return mountNumChanged; }
			
		}
		///<summary>The x position, in pixels, of the item on the mount.</summary>
		[DataField("Xpos")]
		private int xpos;

		/// <summary>The x position, in pixels, of the item on the mount.</summary>
		public int Xpos {
			get { return xpos; }
			set { xpos = value; MarkDirty(); xposChanged = true; }
		}

		bool xposChanged;

		public bool XposChanged {
			get { return xposChanged; }
			
		}
		///<summary>The y position, in pixels, of the item on the mount.</summary>
		[DataField("Ypos")]
		private int ypos;

		/// <summary>The y position, in pixels, of the item on the mount.</summary>
		public int Ypos {
			get { return ypos; }
			set { ypos = value; MarkDirty(); yposChanged = true; }
		}

		bool yposChanged;

		public bool YposChanged {
			get { return yposChanged; }
			
		}
		///<summary>The ordinal position of the item on the mount.</summary>
		[DataField("OrdinalPos")]
		private int ordinalPos;

		/// <summary>The ordinal position of the item on the mount.</summary>
		public int OrdinalPos {
			get { return ordinalPos; }
			set { ordinalPos = value; MarkDirty(); ordinalPosChanged = true; }
		}

		bool ordinalPosChanged;

		public bool OrdinalPosChanged {
			get { return ordinalPosChanged; }
			
		}
		///<summary>The scaled or unscaled width of the mount item in pixels.</summary>
		[DataField("Width")]
		private int width;

		/// <summary>The scaled or unscaled width of the mount item in pixels.</summary>
		public int Width {
			get { return width; }
			set { width = value; MarkDirty(); widthChanged = true; }
		}

		bool widthChanged;

		public bool WidthChanged {
			get { return widthChanged; }
			
		}
		///<summary>The scaled or unscaled height of the mount item in pixels.</summary>
		[DataField("Height")]
		private int height;

		/// <summary>The scaled or unscaled height of the mount item in pixels.</summary>
		public int Height {
			get { return height; }
			set { height = value; MarkDirty(); heightChanged = true; }
		}

		bool heightChanged;

		public bool HeightChanged {
			get { return heightChanged; }
			
		}

		///<summary></summary>
		public MountItem Copy() {
			MountItem m=new MountItem();
			m.MountItemNum=MountItemNum;
			m.MountNum=MountNum;
			m.Xpos=Xpos;
			m.Ypos=Ypos;
			m.OrdinalPos=OrdinalPos;
			m.Width=Width;
			m.Height=Height;
			return m;
		}

	}
}

File: OpenDentalWeb/OpenDental/src/com/opendental/odweb/client/tabletypes/MountItem.java
package com.opendental.odweb.client.tabletypes;

import com.google.gwt.xml.client.Document;
import com.opendental.odweb.client.remoting.Serializing;

//DO NOT MAKE CHANGES TO THIS FILE.  THEY WILL GET OVERWRITTEN BY THE CRUD.
public class MountItem {
		/** Primary key. */
		public int MountItemNum;
		/** FK to mount.MountNum. */
		public int MountNum;
		/** The x position, in pixels, of the item on the mount. */
		public int Xpos;
		/** The y position, in pixels, of the item on the mount. */
		public int Ypos;
		/** The ordinal position of the item on the mount. */
		public int OrdinalPos;
		/** The scaled or unscaled width of the mount item in pixels. */
		public int Width;
		/** The scaled or unscaled height of the mount item in pixels. */
		public int Height;

		/** Deep copy of object. */
		public MountItem deepCopy() {
			MountItem mountitem=new MountItem();
			mountitem.MountItemNum=this.MountItemNum;
			mountitem.MountNum=this.MountNum;
			mountitem.Xpos=this.Xpos;
			mountitem.Ypos=this.Ypos;
			mountitem.OrdinalPos=this.OrdinalPos;
			mountitem.Width=this.Width;
			mountitem.Height=this.Height;
			return mountitem;
		}

		/** Serialize the object into XML. */
		public String serialize() {
			StringBuilder sb=new StringBuilder();
			sb.append("<MountItem>");
			sb.append("<MountItemNum>").append(MountItemNum).append("</MountItemNum>");
			sb.append("<MountNum>").append(MountNum).append("</MountNum>");
			sb.append("<Xpos>").append(Xpos).append("</Xpos>");
			sb.append("<Ypos>").append(Ypos).append("</Ypos>");
			sb.append("<OrdinalPos>").append(OrdinalPos).append("</OrdinalPos>");
			sb.append("<Width>").append(Width).append("</Width>");
			sb.append("<Height>").append(Height).append("</Height>");
			sb.append("</MountItem>");
			return sb.toString();
		}

		/** Sets all the variables on this object based on the values in the XML document.  Variables that are not in the XML document will be null or their default values.
		 * @param doc A parsed XML document.  Must be valid XML.  Does not need to contain a node for every variable on this object.
		 * @throws Exception DeserializeFromXml is entirely encased in a try catch and will throw exceptions if anything goes wrong. */
		public void deserialize(Document doc) throws Exception {
			try {
				if(Serializing.getXmlNodeValue(doc,"MountItemNum")!=null) {
					MountItemNum=Integer.valueOf(Serializing.getXmlNodeValue(doc,"MountItemNum"));
				}
				if(Serializing.getXmlNodeValue(doc,"MountNum")!=null) {
					MountNum=Integer.valueOf(Serializing.getXmlNodeValue(doc,"MountNum"));
				}
				if(Serializing.getXmlNodeValue(doc,"Xpos")!=null) {
					Xpos=Integer.valueOf(Serializing.getXmlNodeValue(doc,"Xpos"));
				}
				if(Serializing.getXmlNodeValue(doc,"Ypos")!=null) {
					Ypos=Integer.valueOf(Serializing.getXmlNodeValue(doc,"Ypos"));
				}
				if(Serializing.getXmlNodeValue(doc,"OrdinalPos")!=null) {
					OrdinalPos=Integer.valueOf(Serializing.getXmlNodeValue(doc,"OrdinalPos"));
				}
				if(Serializing.getXmlNodeValue(doc,"Width")!=null) {
					Width=Integer.valueOf(Serializing.getXmlNodeValue(doc,"Width"));
				}
				if(Serializing.getXmlNodeValue(doc,"Height")!=null) {
					Height=Integer.valueOf(Serializing.getXmlNodeValue(doc,"Height"));
				}
			}
			catch(Exception e) {
				throw e;
			}
		}


}
